src-java/kilda-model/src/main/java/org/openkilda/adapter/FlowDestAdapter.java

```diff
@@ -18,13 +18,18 @@
 import org.openkilda.model.DetectConnectedDevices;
 import org.openkilda.model.Flow;
 import org.openkilda.model.FlowEndpoint;
+import org.openkilda.model.FlowPath;
 import org.openkilda.model.PathId;
 
 import lombok.NonNull;
 
 public class FlowDestAdapter extends FlowSideAdapter {
     public FlowDestAdapter(Flow flow) {
-        super(flow);
+        super(flow, flow.getForwardPath());
+    }
+
+    public FlowDestAdapter(Flow flow, FlowPath flowPath) {
```

Comment:
It ruins the original idea - now you can't have getters like `isPrimaryEgressPath`.

Reply:
Original idea restored. Multitable related properties removed from adapters.